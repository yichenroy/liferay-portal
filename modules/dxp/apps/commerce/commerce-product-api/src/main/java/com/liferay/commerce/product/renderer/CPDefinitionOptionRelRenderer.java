/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 * <p>
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 * <p>
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.renderer;

import aQute.bnd.annotation.ProviderType;

import com.liferay.portal.kernel.exception.PortalException;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

/**
 * @author Marco Leo
 */
@ProviderType
public interface CPDefinitionOptionRelRenderer {

	public String render(
		long cpDefinitionId, HttpServletRequest httpServletRequest,
		HttpServletResponse httpServletResponse) throws PortalException;

}