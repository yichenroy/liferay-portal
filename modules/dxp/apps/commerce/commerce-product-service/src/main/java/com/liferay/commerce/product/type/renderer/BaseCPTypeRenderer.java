/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.type.renderer;

import com.liferay.commerce.product.type.CPTypeRenderer;
import com.liferay.portal.kernel.log.Log;
import com.liferay.portal.kernel.log.LogFactoryUtil;
import com.liferay.portal.kernel.model.PortletConstants;
import com.liferay.portal.kernel.portlet.PortletBag;
import com.liferay.portal.kernel.portlet.PortletBagPool;
import com.liferay.portal.kernel.util.Portal;
import com.liferay.portal.kernel.util.Validator;
import com.liferay.portal.kernel.util.WebKeys;

import java.io.IOException;

import javax.servlet.RequestDispatcher;
import javax.servlet.ServletContext;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.osgi.service.component.annotations.Reference;

/**
 * @author Marco Leo
 */
public abstract class BaseCPTypeRenderer implements CPTypeRenderer {

	public void renderJSP(
			HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse, String path)
		throws IOException {

		renderJSP(
			getServletContext(httpServletRequest), httpServletRequest,
			httpServletResponse, path);
	}

	public void renderJSP(
			ServletContext servletContext,
			HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse, String path)
		throws IOException {

		RequestDispatcher requestDispatcher =
			servletContext.getRequestDispatcher(path);

		try {
			requestDispatcher.include(httpServletRequest, httpServletResponse);
		}
		catch (ServletException se) {
			_log.error("Unable to render JSP " + path, se);

			throw new IOException("Unable to render " + path, se);
		}
	}

	public void setServletContext(ServletContext servletContext) {
		_servletContext = servletContext;
	}

	protected ServletContext getServletContext(
		HttpServletRequest httpServletRequest) {

		if (_servletContext != null) {
			return _servletContext;
		}

		String portletId = portal.getPortletId(httpServletRequest);

		if (Validator.isNotNull(portletId)) {
			String rootPortletId = PortletConstants.getRootPortletId(portletId);

			PortletBag portletBag = PortletBagPool.get(rootPortletId);

			return portletBag.getServletContext();
		}

		return (ServletContext)httpServletRequest.getAttribute(WebKeys.CTX);
	}

	@Reference
	protected Portal portal;

	private static final Log _log = LogFactoryUtil.getLog(
		BaseCPTypeRenderer.class);

	private ServletContext _servletContext;

}