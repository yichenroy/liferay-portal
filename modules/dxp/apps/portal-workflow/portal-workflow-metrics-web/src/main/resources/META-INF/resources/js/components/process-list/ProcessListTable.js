import ProcessListItem from './ProcessListItem';
import React from 'react';

/**
 * @class
 * @memberof processes-list
 * */
export default class ProcessListTable extends React.Component {
	render() {
		const {items} = this.props;

		return (
			<div className="table-responsive">
				<table className="show-quick-actions-on-hover table table-autofit table-heading-nowrap table-hover table-list">
					<thead>
						<tr>
							<th
								className="table-cell-expand table-head-title"
								style={{width: '70%'}}
							>
								{Liferay.Language.get('process-name')}
							</th>
							<th
								className="table-cell-expand table-head-title"
								style={{width: '15%'}}
							>
								{Liferay.Language.get('total-open')}
							</th>
							<th
								className="table-cell-expand table-head-title"
								style={{width: '15%'}}
							>
								{Liferay.Language.get('on-time')}
							</th>
							<th
								className="table-cell-expand table-head-title"
								style={{width: '15%'}}
							>
								{Liferay.Language.get('overdue')}
							</th>
						</tr>
					</thead>

					<tbody>
						{items.map((process, index) => (
							<ProcessListItem {...process} key={index} />
						))}
					</tbody>
				</table>
			</div>
		);
	}
}