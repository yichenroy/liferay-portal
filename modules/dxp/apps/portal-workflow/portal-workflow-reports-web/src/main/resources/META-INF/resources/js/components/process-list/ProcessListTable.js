import ProcessListItem from './ProcessListItem';
import React from 'react';

export default class ProcessListTable extends React.Component {
	render() {
		const {processes} = this.props;

		return (
			<div className="table-responsive">
				<table className="show-quick-actions-on-hover table table-autofit table-hover table-list table-nowrap">
					<thead>
						<tr>
							<th
								className="table-cell-expand table-head-title"
								style={{width: '70%'}}
							>
								{Liferay.Language.get('process-name')}
							</th>
							<th
								className="table-cell-expand table-head-title"
								style={{width: '15%'}}
							>
								{Liferay.Language.get('process-total-open')}
							</th>
							<th
								className="table-cell-expand table-head-title"
								style={{width: '15%'}}
							>
								{Liferay.Language.get('process-on-time')}
							</th>
							<th
								className="table-cell-expand table-head-title"
								style={{width: '15%'}}
							>
								{Liferay.Language.get('process-overdue')}
							</th>
						</tr>
					</thead>

					<tbody>
						{processes.map(({instancesCount, title}, index) => (
							<ProcessListItem
								instancesCount={instancesCount}
								key={index}
								onTime="-"
								overdue="-"
								processName={title}
							/>
						))}
					</tbody>
				</table>
			</div>
		);
	}
}