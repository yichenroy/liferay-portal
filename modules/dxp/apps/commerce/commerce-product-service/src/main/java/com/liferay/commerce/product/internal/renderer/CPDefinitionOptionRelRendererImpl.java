/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.internal.renderer;

import com.liferay.commerce.product.model.CPDefinitionOptionRel;
import com.liferay.commerce.product.model.CPDefinitionOptionValueRel;
import com.liferay.commerce.product.renderer.CPDefinitionOptionRelRenderer;
import com.liferay.commerce.product.service.CPDefinitionOptionRelService;
import com.liferay.dynamic.data.mapping.form.renderer.DDMFormRenderer;
import com.liferay.dynamic.data.mapping.form.renderer.DDMFormRenderingContext;
import com.liferay.dynamic.data.mapping.model.DDMForm;
import com.liferay.dynamic.data.mapping.model.DDMFormField;
import com.liferay.dynamic.data.mapping.model.DDMFormFieldOptions;
import com.liferay.dynamic.data.mapping.model.LocalizedValue;
import com.liferay.portal.kernel.exception.PortalException;
import com.liferay.portal.kernel.util.Portal;

import java.util.List;
import java.util.Locale;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.osgi.service.component.annotations.Component;
import org.osgi.service.component.annotations.Reference;

/**
 * @author Marco Leo
 */
@Component(immediate = true)
public class CPDefinitionOptionRelRendererImpl
	implements CPDefinitionOptionRelRenderer {

	@Override
	public String render(
			long cpDefinitionId, HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse)
		throws PortalException {

		DDMForm ddmForm = new DDMForm();

		Locale locale = _portal.getLocale(httpServletRequest);

		List<CPDefinitionOptionRel> cpDefinitionOptionRels =
			_cpDefinitionOptionRelService.
				getSkuContributorCPDefinitionOptionRels(cpDefinitionId);

		for (CPDefinitionOptionRel cpDefinitionOptionRel :
				cpDefinitionOptionRels) {

			List<CPDefinitionOptionValueRel> cpDefinitionOptionValueRels =
				cpDefinitionOptionRel.getCPDefinitionOptionValueRels();

			DDMFormField ddmFormField = new DDMFormField(
				String.valueOf(cpDefinitionOptionRel.
					getCPDefinitionOptionRelId()),
				cpDefinitionOptionRel.getDDMFormFieldTypeName());

			if (!cpDefinitionOptionValueRels.isEmpty()) {
				DDMFormFieldOptions ddmFormFieldOptions =
					new DDMFormFieldOptions();

				for (CPDefinitionOptionValueRel cpDefinitionOptionValueRel :
						cpDefinitionOptionValueRels) {

					ddmFormFieldOptions.addOptionLabel(
						String.valueOf(cpDefinitionOptionValueRel.
							getCPDefinitionOptionValueRelId()),
						locale, cpDefinitionOptionValueRel.getTitle(locale));
				}

				ddmFormField.setDDMFormFieldOptions(ddmFormFieldOptions);
			}

			LocalizedValue localizedValue = new LocalizedValue(locale);

			localizedValue.addString(
				locale, cpDefinitionOptionRel.getName(locale));

			ddmFormField.setLabel(localizedValue);

			ddmFormField.setRequired(true);

			ddmForm.addDDMFormField(ddmFormField);
		}

		DDMFormRenderingContext ddmFormRenderingContext =
			new DDMFormRenderingContext();

		ddmFormRenderingContext.setContainerId("options");
		ddmFormRenderingContext.setHttpServletRequest(httpServletRequest);
		ddmFormRenderingContext.setHttpServletResponse(httpServletResponse);
		ddmFormRenderingContext.setLocale(locale);

		return _ddmFormRenderer.render(ddmForm, ddmFormRenderingContext);
	}

	@Reference
	private CPDefinitionOptionRelService _cpDefinitionOptionRelService;

	@Reference
	private DDMFormRenderer _ddmFormRenderer;

	@Reference
	private Portal _portal;

}