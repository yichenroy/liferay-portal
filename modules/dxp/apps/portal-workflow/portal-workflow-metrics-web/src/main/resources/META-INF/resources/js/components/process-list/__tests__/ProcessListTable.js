import ProcessListTable from '../ProcessListTable';
import React from 'react';
import renderer from 'react-test-renderer';
import { MockRouter as Router } from '../../../test/mock/MockRouter';

test('Should render component', () => {
	const data = [
		{
			id: 36401,
			instancesCount: 0,
			title: 'test'
		}
	];

	const component = renderer.create(
		<Router>
			<ProcessListTable items={data} />
		</Router>
	);

	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});