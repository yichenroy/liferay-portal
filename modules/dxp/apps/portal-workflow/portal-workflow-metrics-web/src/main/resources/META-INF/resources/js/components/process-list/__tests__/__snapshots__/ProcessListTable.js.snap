// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render component 1`] = `
<div
  className="table-responsive"
>
  <table
    className="show-quick-actions-on-hover table table-autofit table-heading-nowrap table-hover table-list"
  >
    <thead>
      <tr>
        <th
          className="table-cell-expand table-head-title"
          style={
            Object {
              "width": "70%",
            }
          }
        >
          <a
            className="inline-item text-truncate-inline"
            href="/?backPath=%2F"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
              data-title="Process Name"
              title="Process Name"
            >
              Process Name
            </span>
            <span
              className="inline-item inline-item-after"
            >
              <svg
                className="lexicon-icon lexicon-icon-order-arrow-down"
                focusable="false"
                role="presentation"
              >
                <use
                  href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#order-arrow-down"
                />
              </svg>
            </span>
          </a>
        </th>
        <th
          className="table-cell-expand table-head-title"
          style={
            Object {
              "width": "15%",
            }
          }
        >
          <a
            className="inline-item text-truncate-inline"
            href="/?backPath=%2F"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
              data-title="Overdue"
              title="Overdue"
            >
              Overdue
            </span>
            <span
              className="inline-item inline-item-after"
            >
              <svg
                className="lexicon-icon lexicon-icon-order-arrow-down"
                focusable="false"
                role="presentation"
              >
                <use
                  href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#order-arrow-down"
                />
              </svg>
            </span>
          </a>
        </th>
        <th
          className="table-cell-expand table-head-title"
          style={
            Object {
              "width": "15%",
            }
          }
        >
          <a
            className="inline-item text-truncate-inline"
            href="/?backPath=%2F"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
              data-title="On Time"
              title="On Time"
            >
              On Time
            </span>
            <span
              className="inline-item inline-item-after"
            >
              <svg
                className="lexicon-icon lexicon-icon-order-arrow-down"
                focusable="false"
                role="presentation"
              >
                <use
                  href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#order-arrow-down"
                />
              </svg>
            </span>
          </a>
        </th>
        <th
          className="table-cell-expand table-head-title"
          style={
            Object {
              "width": "15%",
            }
          }
        >
          <a
            className="inline-item text-truncate-inline"
            href="/?backPath=%2F"
            onClick={[Function]}
          >
            <span
              className="text-truncate"
              data-title="Total Pending"
              title="Total Pending"
            >
              Total Pending
            </span>
            <span
              className="inline-item inline-item-after"
            >
              <svg
                className="lexicon-icon lexicon-icon-order-arrow-down"
                focusable="false"
                role="presentation"
              >
                <use
                  href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#order-arrow-down"
                />
              </svg>
            </span>
          </a>
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="table-cell-expand table-cell-minw-200 table-title lfr-title-column"
        >
          <a
            href="/process-dashboard/36401?backPath=%2Fprocesses%2F1%2F10%2Ftitle%253Aasc%3FbackPath%3D%252F"
            onClick={[Function]}
          >
            <span>
              test
            </span>
          </a>
        </td>
        <td>
          -
        </td>
        <td>
          -
        </td>
        <td>
          -
        </td>
        <td>
          <div
            className="dropdown dropdown-action"
          >
            <a
              aria-expanded="false"
              aria-haspopup="true"
              className="component-action dropdown-toggle"
              data-toggle="dropdown"
              href="#1"
              id="dropdownAction1"
              role="button"
            >
              <svg
                className="lexicon-icon lexicon-icon-ellipsis-v"
                focusable="false"
                role="presentation"
              >
                <use
                  href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#ellipsis-v"
                />
              </svg>
            </a>
            <ul
              aria-labelledby=""
              className="dropdown-menu dropdown-menu-right"
            >
              <li>
                <a
                  className="dropdown-item"
                  href="/slas/36401/20/1?backPath=%2Fprocesses%2F1%2F10%2Ftitle%253Aasc%3FbackPath%3D%252F"
                  onClick={[Function]}
                >
                  Set Up SLAs
                </a>
              </li>
            </ul>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
